pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -418,6 +419,82 @@ func TestReconciler(t *testing.T) {
 				Image("", nil).
 				Obj(),
 		},
+		"when workload is evicted, suspend, reset startTime and restore node affinity": {
+			job: *utiltestingjob.MakeJob("job", "ns").
+				Suspend(false).
+				StartTime(time.Now()).
+				NodeSelector("provisioning", "spot").
+				Parallelism(10).
+				Request(corev1.ResourceCPU, "1").
+				Image("", nil).
+				Active(10).
+				Obj(),
+			workloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("a", "ns").
+					PodSets(*utiltesting.MakePodSet("main", 10).Request(corev1.ResourceCPU, "1").Obj()).
+					Admit(utiltesting.MakeAdmission("cq").AssignmentPodCount(10).Obj()).
+					Condition(metav1.Condition{
+						Type:   kueue.WorkloadEvicted,
+						Status: metav1.ConditionTrue,
+					}).
+					Obj(),
+			},
+			wantJob: *utiltestingjob.MakeJob("job", "ns").
+				Suspend(true).
+				Parallelism(10).
+				Request(corev1.ResourceCPU, "1").
+				Image("", nil).
+				Active(10).
+				Obj(),
+			wantWorkloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("a", "ns").
+					PodSets(*utiltesting.MakePodSet("main", 10).Request(corev1.ResourceCPU, "1").Obj()).
+					Admit(utiltesting.MakeAdmission("cq").AssignmentPodCount(10).Obj()).
+					Condition(metav1.Condition{
+						Type:   kueue.WorkloadEvicted,
+						Status: metav1.ConditionTrue,
+					}).
+					Obj(),
+			},
+		},
+		"when workload is evicted but suspended, reset startTime and restore node affinity": {
```

Comment:
I would suggest to add more test case: "when workload is evicted, but suspended, and the startTime is rest, restore node affinity"

Reply:
added.